components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java

```diff
@@ -104,8 +109,42 @@ private OIDCClaimUtil() {
     public static String getServiceProviderMappedUserRoles(ServiceProvider serviceProvider,
                                                            List<String> locallyMappedUserRoles,
                                                            String claimSeparator) throws FrameworkException {
+
         if (isNotEmpty(locallyMappedUserRoles)) {
             locallyMappedUserRoles = new ArrayList<>(locallyMappedUserRoles);
+            /*
+             Check whether the application is a fragment application and if so, add only the shared roles
+             to the user's role list.
+            */
+            if (serviceProvider.getSpProperties() != null && Arrays.stream(serviceProvider.getSpProperties()).anyMatch(
```

Comment:
Since this method is used to resolve service provider mapped user roles, shall we look into the place where `locallyMappedUserRoles` are resolved and the possibility of adding this logic to that place.